src/TextareaAutosize.js

```diff
@@ -87,6 +93,13 @@ export default class TextareaAutosize extends React.Component {
     // Re-render with the new content then recalculate the height as required.
     this.onNextFrameActionId = onNextFrame(this._resizeComponent);
   }
+  
+  componentWillUpdate(nextProps, nextState) {
```

Comment:
@Andreyco I'm more about the fact that `setState` is not allowed from within the `componentWillUpdate` so it is interesting, what if parent is updated from there and re-rendered its children including this component.